src/Util/ColumnParser.php

```diff
@@ -16,14 +16,14 @@ class ColumnParser
      *
      * @var string
      */
-    protected $regexpParseColumn = '/^(\w*)(?::(\w*\??\[?\d*\]?))?(?::(\w*))?(?::(\w*))?/';
+    protected $regexpParseColumn = '/^(\w*)(?::(\w*\??\[?(?:\d|[0-9,])*\]?))?(?::(\w*))?(?::(\w*))?/';
```

Comment:
Do we need to allow values like `decimal[10,10,10,10]` or could the pattern be more strict?

Reply:
The documentation should handle it?

Currently it even accepts something like `amount:string[0]:primary`.

Actually I tried it...
`$regexpParseColumn` governs many things and when I tried adding groups, it created matches with blanks because some were satisfiable and some weren't. For instance, when I use a pipe to signify OR, I will have create 2 distinct groups to make it more strict, which is currently summarized as `?:\d|[0-9],`.

We cannot get any more strict without having to add more groups and that complicates the other functions which relies on this variable. For install `return [$matches[1], $matches[2]];` which uses the index of the matched expression. By adding more groups, I had to create more and more `if - else` at various places and things got a little bit ugly, so I decided to keep validation a bit more relaxed; trusting that user will know what decimal is and will read the docs.

What do you reckon?

Reply:
My concern is that they'll end up with broken SQL generated and have a hard time tracing that back to invalid input. Could we check the matches to ensure there is only one `,` as increasing complexity in the patterns will have ripples elsewhere.

Reply:
Oh dear... Well, ok... Let me see what regex I can come up with... Cool. on it.

Reply:
In case it's of any help, I tested how to make things a little more strict when I had a look at the nullable thing. That's what I had so far:

```regex
/^(\w+)(?::(\w+\??(?:\[(?:[0-9]|[1-9][0-9]+)(?:,(?:[0-9]|[1-9][0-9]+))?\])?))?(?::(\w+))?(?::(\w+))?$/
```

Reply:
Hang on a second... I remember list handling this... there will be no wrongly generated SQL because it will accept only first 2 instances. One for precision and other for scale.

Reply:
@ndm2 Are you advising for the whole or just decimals? Because I am using `list` to extract precision and scale. The idea is to not generate wrong SQL. Line 67.
Yours is better though... https://regexr.com/4hqsf and https://regexr.com/4hqso